### PR TITLE
Adding an ignore to allow for -- syntax on a modifier without an error

### PR DESCRIPTION
**Pull request by Philipa1**
Ticket: https://webjobs.agepartnership.co.uk/desk/tickets/7163252/messages

Part of: https://github.com/AgePartnership/oleg-property-wealth-brand-kit/pull/2#discussion_r1693021311

Change: Added an ignore to allow for -- syntax on a modifier without an error 

Checks: malice, sense, ISO

### CODE REVIEW

**Review by kimoshinowo98 — COMMENTED** [review]
As far as I can see, the only two options for this are `local` or `global` (https://github.com/stylelint-scss/stylelint-scss/blob/master/src/rules/dollar-variable-pattern/README.md#optional-options) - where did `selector-class-pattern` come from? Have you tested on your PW branch to make sure this fixes the issue?

**Comment by Philipa1** [comment]
Oh, didn't see that! I was just following the structure of the other 'ignore' rules as I thought that would fix it. This `selector-class-pattern` is what it is erroring on. @kimoshinowo98

**Comment by Philipa1** [comment]
@kimoshinowo98 Ready for review now when you get a second :)

**Review by kimoshinowo98 — COMMENTED** [review]
Indentation needs fixing but otherwise fine!

**Review by kimoshinowo98 — APPROVED** [review]

File: index.js [hunk 1]
```diff
@@ -89,5 +89,12 @@ export default {
         "selector-max-id": 0,
         "selector-pseudo-element-colon-notation": "double",
         "property-no-unknown": true,
+        "selector-class-pattern": [
+            "^([a-z][a-z0-9]*)(-[a-z0-9]+)*$",
```
Comment (beccadawson) [inline index.js:93]: I'm not sure this is the correct pattern for how we name our classes – do you know how you'd check that?

**Comment by beccadawson** [comment]
@kimoshinowo98 @Philipa1 not sure that my comment will have notified you since this is merged!

File: index.js (hunk 1 above)
Comment (kimoshinowo98) [reply index.js:93]: @beccadawson it's the pattern that's set in the default styelint config https://github.com/stylelint/stylelint-config-standard/blob/main/index.js#L95-L100 so I though it was the same pattern we already use?

Comment (beccadawson) [reply index.js:93]: AFACIT we didn't actually use the standard config prior to v3 (https://github.com/AgePartnership/ap-stylelint-config/tree/v2.1.2), so think we'll actually have added a bunch of additional rules as part of the update to Stylelint 16 (and, sidenote, it looks like we're using the standard CSS config, not the SCSS config... I did a bad job at reviewing this 😅 )

So I think we just didn't have a check on this before!

Comment (kimoshinowo98) [reply index.js:93]: @beccadawson Oooh I did not realise that 😬😅 Shall I remove the use of the standard config (at least for now) then?

Comment (beccadawson) [reply index.js:93]: Yeah maybe 😛 worth having on the radar to revisit our configs though (having a check that we're actually using bem conventions has been on my radar for a whlile!)

**Pull request merged**